sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -146,16 +190,117 @@ public void onChallengeCredentialPolicy() {
+        onChallengeAndClearCacheSync(policy, this.callContext, unauthorizedHttpResponseWithHeader);
         // Challenge cache used
-        policy.authorizeRequest(this.testContext).block();
+        policy.authorizeRequestSync(this.testContext);
 
-        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
         assertFalse(tokenValue.isEmpty());
         assertTrue(tokenValue.startsWith(BEARER));
     }
 
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentWithClaims() {
+        MutableTestCredential testCredential = new MutableTestCredential();
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(testCredential, false);
+
+        StepVerifier.create(policy.authorizeRequestOnChallenge(this.callContext, // Challenge cache created
+                    this.unauthorizedHttpResponseWithHeader)
+                .flatMap(authorized -> {
+                    if (authorized) {
+                        String firstToken = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
+
+                        assertFalse(firstToken.isEmpty());
+                        assertTrue(firstToken.startsWith(BEARER));
+
+                        return policy.authorizeRequestOnChallenge(this.callContext, // Challenge with claims received
+                                this.unauthorizedHttpResponseWithHeaderAndClaims)
+                            .map(ignored -> firstToken);
+                    } else {
+                        return Mono.just("");
+                    }
+                }))
+            .assertNext(firstToken -> {
+                String newToken = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
+
+                assertFalse(newToken.isEmpty());
+                assertTrue(newToken.startsWith(BEARER));
+
+                assertNotEquals(firstToken, newToken);
+            })
+            .verifyComplete();
+
+        KeyVaultCredentialPolicy.clearCache();
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeNoCachePresentWithClaims() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
+        StepVerifier.create(policy.authorizeRequestOnChallenge(this.callContext, // Challenge cache created
+                this.unauthorizedHttpResponseWithHeaderAndClaims))
+            .assertNext(result -> {
+                assertFalse(result);
+                assertNull(this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION));
+            })
+            .verifyComplete();
+
+        KeyVaultCredentialPolicy.clearCache();
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentWithClaimsSync() {
+        MutableTestCredential testCredential = new MutableTestCredential();
```

Comment:
This test credential should verify the policy passes the correct claims and always requests CAE-enabled tokens

Reply:
The test credential checks that CAE is enabled and claims are present in the `TokenRequestContext` object in order to provide a new token. Otherwise, the old token is returned and the test fails as the old token and new token are not different.

If it gives more peace of mind, we can make sure the claims are exactly the same as the decoded String.